### ISSUE TITLE
Requests are send twice

### ISSUE THREAD

**Issue by gitnik** [gund/ng2-http-interceptor#27]
I am running into an issue where every HTTP request is fired twice. All I did was include this library and use the `HTTP_INTERCEPTOR_PROVIDER`. Everything else is standard angular2 http handling. Did anyone else experience this issue as well? If not, I will provide a plunker when I have more time

**Comment by gund** [comment]
I just tried it and also found that behavior the only difference is that in my case I got 3 requests instead of 1.
Seems to be a bug and also tried in versions 0.4.0 and 0.3.0 - also there.

Really interesting how I missed it =)
Will check what is causing those extra requests and thanks for reporting this issue.

**Comment by gitnik** [comment]
I think the issue originates from this 'flapMap' call: https://github.com/gund/ng2-http-interceptor/blob/master/src/http/interceptable-http-proxy.service.ts#L39

It seems that while combining the observables the http method is called again. I only investigated this for 5 minutes though, so I might be totally wrong

**Comment by gund** [comment]
I have added [verification to the tests](https://github.com/gund/ng2-http-interceptor/blob/73ca2056c6c55550800235870787b179b51bf51a/src/http/high-level-api.spec.ts#L127) to check if I do not duplicate any requests when interceptors are registered and [they show](https://travis-ci.org/gund/ng2-http-interceptor/jobs/171143826#L490) that there are no repeated requests whatsoever, which is really strange.

Probably will need to dive deeper what is happening because now it seems weird.

**Comment by gund** [comment]
I finally was able to fix this issue.

The problem was in so called "cold" observables - once a new subscription done on that observable whole new HTTP request done again.
So what I did is internally "share" observable or transform from being "cold" -> to "hot" and return that.

It works fine but maybe somewhat unexpected behavior but this is the only way to resolve it with current API.

I also thinking about changing API for response interceptor functions, so instead of receiving `Observable<Response` they will receive unwrapped `Response` and should return it as well.
However this will be a breaking change
